makelib/lib.mk

```diff
@@ -0,0 +1,93 @@
+# Copyright 2015 The rkt Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# find-so-deps scans the passed folder and returns a list of external *.so libraries
+# that the binaries in this path are depending on
+find-so-deps = $(shell cd $(1) && find -type f | xargs file | grep ELF | cut -f1 -d: | LD_LIBRARY_PATH=usr/lib xargs ldd | grep -v ^\\. | grep '/' | sed  -e 's/^[[:space:]]*//' -e 's/.*=> //' -e 's/ (0x[0-9a-f]*)$$//' | grep -v ^[^/] | sort -u)
+
+# find-file-so-deps returns the list of external *.so libraries used by the binary
+# passed as an argument
+find-file-so-deps = $(shell LD_LIBRARY_PATH=usr/lib ldd $(1) | grep -v ^\\. | grep '/' | sed  -e 's/^[[:space:]]*//' -e 's/.*=> //' -e 's/ (0x[0-9a-f]*)$$//' | grep -v ^[^/] | sort -u)
```

Comment:
The LD_LIBRARY_PATH here should be probably configurable (as a `$(2)` variable). This is used inside directories like stage1/bash, stage1/gc or stage1/init which obviously don't have any usr/lib directory.